cpp/tests/io/parquet_writer_test.cpp

```diff
@@ -185,7 +185,7 @@ TEST_F(ParquetWriterTest, BufferSource)
 TEST_F(ParquetWriterTest, ManyFragments)
 {
   srand(31337);
-  auto const expected = create_random_fixed_table<int>(10, 6'000'000, false);
+  auto const expected = create_random_fixed_table<int>(2, 2'000'000, false);
```

Comment:
I want to check on this one...I was already looking at it. We need to make sure there are enough rows/columns to generate over 64k fragments.

Reply:
do we want 64k fragments per column or in total?

Reply:
Per column. This was meant to test changes made to `gpuInitRowGroupFragments()`.  which has a max grid dimension of `(num_columns, 64k)`.

Reply:
I think we need smaller fragments then. even with the original size, (6M * sizeof(int)) / (8 * 1024) < 3K fragments.

Reply:
Yeah, something's off now. It's still using 5k fragments which are too big for 8k pages.

Reply:
I don't know if that test ever worked :worried:. How about changing it to one column with 700k rows, and then set `max_page_fragment_size(10)` on the builder? That seems to actually trigger the desired behavior.